### ISSUE TITLE
Froala Custom Buttons not working

### ISSUE THREAD

**Issue by Ladylain** [octobercms/october#4903]
- OctoberCMS Build: 464 
- PHP Version: 7.2.19
- Database Engine: MySQL

### Description:
Hello, 

i'm trying to add some personalized buttons to the RichEditor Widget. To help me, i've read many october issues (mainly #2078 ) on this repo, but unfortunately, after many tries, it does not work.

### Steps To Reproduce:

I have made a plugin for my site, and i have added this snippet in the boot() function 

**Plugin.php**
```php
RichEditor::extend(function($widget) {
      $widget->addJs('/plugins/*****/******/assets/js/test.js');
});
```

**test.js**
```js
+function ($) {
  var Plugins = {
      init: function () {
          $.FroalaEditor.DefineIcon('buttonIcon', {NAME: 'star'})
          $.FroalaEditor.RegisterCommand('myButton', {
              title: 'My Button',
              icon: 'buttonIcon',
              undo: true,
              focus: true,
              refreshAfterCallback: true,
              callback: function () {
                  console.log(this.html.get());
              },
              refresh: function ($btn) {
                  console.log(this.selection.element());
              }
          })

          $.oc.richEditorButtons.push('myButton')
          
      }
  }

  $(document).on('render', function () {
      Plugins.init()
  })
}(jQuery)
```

please do not focus on file names, or functions, i'm just trying to add the button first, then i will make my functions. 

Does somebody have any solutions ?

Best, 
L.

**Comment by gergo85** [comment]
This is a working implementation: https://github.com/gergo85/oc-paste-content/blob/master/assets/js/froala.paste.plugin.js

**Comment by Ladylain** [comment]
@gergo85 Thanks for your answer, i've just tried this and it works BUT only if you don't fill in this field in the settings area 

![image](https://user-images.githubusercontent.com/5721178/72980426-da214600-3dda-11ea-8dde-6484f41c75c7.png)

Do you have any solution that works with this configuration ?

Best,

**Comment by gergo85** [comment]
I tried it. Indeed, it is a bug. I have no idea what is the solution.

**Comment by LukeTowers** [comment]
@w20k can you look into this?

**Comment by w20k** [comment]
@LukeTowers sure-sure!

**Comment by alxy** [comment]
Probably you need to add myButton to this config list?

**Comment by Ladylain** [comment]
Sorry but i've already tried this, and it does not fix this

**Comment by Ladylain** [comment]
Hello there, 

is there any infos about this fix ? 

Thx

**Comment by mjauvin** [comment]
@Ladylain look at the source code, that explains why this is not working:

   `modules/backend/formwidgets/richeditor/assets/js/richeditor.js`

When something is defined in settings it gets assigned to `this.options.toolbarButtons`  through the widget's markup.

**Comment by mjauvin** [comment]
maybe extending this in your plugin:
```
$buttons = "your buttons here";
$.extend($.FE.DEFAULTS {toolbarButtons:$buttons, toolbarButtonsXS:$buttons, toolbarButtonsSM:$buttons, toolbarButtonsMD:$buttons});
```

That is a pure guess...

**Comment by daftspunk** [comment]
For anyone stumbling across this recently, this has been addressed as an API in v3.4. The documentation now includes these instructions:

---

## Registering a Custom Button

The following JavaScript code can be used to register a custom button as a command.

```js
oc.richEditorRegisterButton('insertCustomThing', {
    title: 'Insert Something',
    icon: '<i class="icon-star"></i>',
    undo: true,
    focus: true,
    refreshOnCallback: true,
    callback: function () {
        this.html.insert('<strong>My Custom Thing!</strong>');
    }
});
```

Then add the button to the default collection.

```js
oc.richEditorButtons.splice(0, 0, 'insertCustomThing');
```
